### PR TITLE
datepicker selecting previous day for UTC-3 TZ fixed

### PR DESCRIPTION
**Pull request by barshat7**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js [hunk 1]
```diff
@@ -1135,8 +1135,13 @@ if (typeof window.DatePickerWidget === 'undefined') {
      */
 
     setValue(value) {
-      let currDate =  new Date(value);
-
+      let currDate;
+      const dateArray = value.split('-');
+      if (dateArray && dateArray.length == 3) {
+        currDate = new Date(dateArray[0], dateArray[1] - 1, dateArray[2]);
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:1141]: For all date parsing logic, we should rely on OOTB formatting logic which is exposed in af-formatters. @vdua Why can't af-core handle this ?

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

**Comment by rismehta** [comment]
@barshat7 Can you override the today function as per this, https://github.com/adobe/json-formula/blob/main/src/functions.js#L2140 in af2-web-runtime under FunctionRuntime ?

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js (hunk 1 above)
Comment (sakshi-arora1) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:1141]: +1 needs to be handled in af2-web-runtime

Comment (rismehta) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:1141]: The issue is specific to the view layer, @barshat7 is working on the fix.

**Comment by barshat7** [comment]
Created another PR for this issue
https://github.com/adobe/aem-core-forms-components/pull/1168
cc @rismehta @sakshi-arora1

**Pull request closed without merging**